tests/integration/helpers.py

```diff
@@ -176,3 +178,32 @@ def gen_git_test_job_xml(node_label: str):
         </project>
         """
     )
+
+
+async def wait_for(
+    func: typing.Callable[[], typing.Union[typing.Awaitable, typing.Any]],
+    timeout: int = 300,
+    check_interval: int = 10,
+) -> typing.Any:
+    """Wait for function execution to become truthy.
+
+    Args:
+        func: A callback function to wait to return a truthy value.
+        timeout: Time in seconds to wait for function result to become truthy.
+        check_interval: Time in seconds to wait between ready checks.
+
+    Raises:
+        TimeoutError: if the callback function did not return a truthy value within timeout.
+
+    Returns:
+        The result of the function if any.
+    """
+    deadline = time.time() + timeout
+    is_awaitable = inspect.iscoroutinefunction(func)
+    while time.time() < deadline:
+        if is_awaitable and (result := await func()):
+            return result
+        if result := func():
+            return result
+        time.sleep(check_interval)
+    raise TimeoutError()
```

Comment:
Do you think that the func should run one more time since `sleep` is the last thing in the loop essentially we would pay the price of the `sleep` without getting a reward in the form of one more try?

Reply:
Added one more check, thank you!